executor/executor_bsd.h

```diff
@@ -121,11 +121,12 @@ static void cover_mmap(cover_t* cov)
 static void cover_protect(cover_t* cov)
 {
 #if GOOS_freebsd
-	size_t mmap_alloc_size = kCoverSize * KCOV_ENTRY_SIZE;
-	long page_size = sysconf(_SC_PAGESIZE);
-	if (page_size > 0)
-		mprotect(cov->data + page_size, mmap_alloc_size - page_size,
-			 PROT_READ);
+	if (cov->data) {
```

Comment:
I wonder in what case we might invoke `cover_protect` on a `cov` that has `cov->data == NULL`.

Reply:
I can implement this in `CoverAccessScope` too. However, it'd be nice for `cover_protect()` to fail closed if it's somehow called while coverage is disabled. Either my patch, or some check `if (cov->data == NULL) failmsg("...");`.

Reply:
> However, it'd be nice for cover_protect() to fail closed if it's somehow called while coverage is disabled.

I totally agree.
Let's go the way of adding the `failmsg` and improving `CoverAccessScope`.

Reply:
Done!